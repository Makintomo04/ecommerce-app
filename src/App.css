.container {
  max-width: 1200px;
  margin: 0 auto;
  height: auto;
  /* margin-top: 80px; */
  /* display: flex;
  flex-direction: column;
  align-items: center; */
}
