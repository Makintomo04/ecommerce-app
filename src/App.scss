.container {
  max-width: 1200px;
  margin: 0 auto;
  margin-top: 200px;
  height: auto;
  // /* margin-top: 80px; */
  display: flex;
  flex-direction: column;
  align-items: center;
}
@media (max-width: 1240px) {
  .container {
    max-width: 780px;
  }
}
@media (max-width: 540px) {
  .container {
    margin-top: 150px;
  }
}
.App {
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  &__Content {
    flex: 1;
  }
}
body,
html {
  width: 100%;
  font-family: "Roboto", sans-serif;
  // background: white;
  /* background: #dcd8cd; */
  /* background: black; */
  overflow-x: hidden;
  // overflow-y: scroll;
}
.home {
  visibility: hidden;
}
.womens-page {
  visibility: hidden;
}
.mens-page {
  visibility: hidden;
}
