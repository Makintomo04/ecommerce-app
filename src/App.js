import React, { useState, useEffect, useRef } from "react";
import "./App.scss";
import LoginPage from "./pages/sign-in-sign-up/SignInSignUp";
import HomePage from "./pages/homepage/HomePage";
import MensPage from "./pages/menspage/MensPage";
import WomensPage from "./pages/womenspage/WomensPage";
import ShoesPage from "./pages/shoespage/Shoes";
import AccessoriesPage from "./pages/accessoriespage/Accessories";
import ActivewearPage from "./pages/activewearpage/Activewear";
import CheckoutPage from "./pages/checkoutpage/Checkout";
import WishListPage from "./pages/wishlistpage/WishList";
import Bag from "./pages/bagpage/BagPage";
import Directory from "./components/directory/Directory";
import { Route, Switch, Redirect } from "react-router-dom";
import Header from "./components/header/Header";
import Footer from "./components/footer/Footer";
import { auth, createUserProfileDocument } from "./firebase/firebase.utils";
import { connect } from "react-redux";
import { setCurrentUser } from "./redux/user/user.actions";
import { createStructuredSelector } from "reselect";
import { selectCurrentUser } from "./redux/user/user.selectors";

class App extends React.Component {
  unsubscribeFromAuth = null;

  componentDidMount() {
    const { setCurrentUser } = this.props;

    this.unsubscribeFromAuth = auth.onAuthStateChanged(async (userAuth) => {
      if (userAuth) {
        const userRef = await createUserProfileDocument(userAuth);

        userRef.onSnapshot((snapShot) => {
          setCurrentUser({
            id: snapShot.id,
            ...snapShot.data(),
          });
        });
      }

      setCurrentUser(userAuth);
    });
  }

  componentWillUnmount() {
    this.unsubscribeFromAuth();
  }
  render() {
    return (
      <div className="App">
        <div className="App__Content">
          <Header />
          <Switch>
            <Route exact path="/" component={HomePage} />
            <Route
              exact
              path="/login"
              render={() =>
                this.props.currentUser ? <Redirect to="/" /> : <LoginPage />
              }
            />
            <Route exact path="/mens" component={MensPage} />
            <Route exact path="/womens" component={WomensPage} />
            <Route path="/shoes" component={ShoesPage} />
            <Route path="/accessories" component={AccessoriesPage} />
            <Route path="/activewear" component={ActivewearPage} />
            <Route exact path="/checkout" component={CheckoutPage} />
            <Route exact path="/bag" component={Bag} />
            <Route exact path="/wishlist" component={WishListPage} />
          </Switch>
        </div>
        {/* <HomePage /> */}
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
});
const mapDispatchToProps = (dispatch) => ({
  setCurrentUser: (user) => dispatch(setCurrentUser(user)),
});
export default connect(mapStateToProps, mapDispatchToProps)(App);
