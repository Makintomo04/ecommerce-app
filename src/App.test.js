import React from "react"
import {shallow} from "enzyme"
import {App} from "./App"


it("Should render App component correctly", () => {
    const app = shallow(<App />);
    expect.assertions(1)
    expect(app.props().children).toMatchSnapshot() // check if a component renders correctly
  });