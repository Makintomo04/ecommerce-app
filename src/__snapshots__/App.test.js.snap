// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App component correctly 1`] = `
Array [
  <div
    className="App__Content"
  >
    <withRouter(Connect(Header)) />
    <Switch>
      <ErrorBoundary>
        <React.Suspense
          fallback={<Spinner />}
        >
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/"
          />
          <Route
            exact={true}
            path="/login"
            render={[Function]}
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/mens"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/mens/new-in"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/mens/shoes"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/mens/accessories"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/mens/activewear"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/womens"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/womens/new-in"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/womens/clothing"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/mens/clothing"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/womens/shoes"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/womens/accessories"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/womens/activewear"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/clothing"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/shoes"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/accessories"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            path="/activewear"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/checkout"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/bag"
          />
          <Route
            component={
              Object {
                "$$typeof": Symbol(react.lazy),
                "_ctor": [Function],
                "_result": null,
                "_status": -1,
              }
            }
            exact={true}
            path="/wishlist"
          />
        </React.Suspense>
      </ErrorBoundary>
    </Switch>
  </div>,
  <Footer />,
]
`;
